### DISCUSSION TITLE
MsSqlContainer UntilContainerIsHealthy wait strategy is not working for me

### DISCUSSION THREAD

**Discussion by cezar-pimentel** [testcontainers/testcontainers-dotnet#1167] [Q&A] [answered]
Hello guys, good evening.

Well, I really looked everywhere, and I could not find anything about how to use the UntilContainerIsHealthy with the MsSqlContainer.

My integration test is failing because some database calls are being triggered before the container is indeed ready to accept calls, even though I'm waiting for the port to be ready.

Can someone provide me a sample or any advice on how to correctly wait for some sort of healthy or startup completion state?

This is how I'm currently configuring my MsSqlContainer:

![image](https://github.com/testcontainers/testcontainers-dotnet/assets/12258489/a4ec6cd3-50d2-4b5e-a9aa-278a42d54bbd)

Because I'm using the **_UntilContainerIsHealthy_** strategy the **_StartAsync_** method never finishes.

Thanks for all the help in advance.

Cheers!

**Comment by HofmeisterAn** [comment, marked as answer]
These wait strategies won't work with MSSQL. 1) The port becomes available before the actual service. 2) AFAIK the image does not provide a [HEALTHCHECK](https://docs.docker.com/reference/dockerfile/#healthcheck). Why don't you use the [MSSQL](https://www.nuget.org/packages/Testcontainers.MsSql) module?

    MsSqlContainer _msSqlContainer = new MsSqlBuilder().Build();

Modules are preconfigured and contain proven best practices. The MSSQL module uses the following [wait strategy](https://github.com/testcontainers/testcontainers-dotnet/blob/91bc71adf2e0bcfa1a52841ae15f22a5f70d1cb9/src/Testcontainers.MsSql/MsSqlBuilder.cs#L132-L143).

**Comment by cezar-pimentel** [comment]
Using the preconfigured wait strategy solved for me! Thanks for the explanation man!

Have a great week!

👊💪
